modules/dkan/dkan_datastore/psr-4/Resource.php

```diff
@@ -69,11 +77,63 @@ public static function createFromDrupalNodeNid($nid) {
    *   Resource.
    */
   public static function createFromDrupalNode($node) {
+    // Return false if wrong type.
+    if ($node->type != self::resourceContentType()) {
+      return FALSE;
+    }
     $id = $node->nid;
     $file_path = self::filePath($node);
     return new self($id, $file_path);
   }
 
+  /**
```

Comment:
These following four functions may not be necessary. This is to preserve legacy functionality that allows the content type and fields used with the datastore to be completely switched if someone wants to use the datastore with a completely different module. In theory I think this is good, but it may be simpler to strip out for now while we're in MVP mode. Also, even if we keep it I don't think these alter hooks are the most elegant way to allow customization -- just a variable would probably make more sense.